eclipse-collections/src/main/java/org/eclipse/collections/impl/list/IntervalUtils.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2019 Goldman Sachs and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompany this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+
+package org.eclipse.collections.impl.list;
+
+public final class IntervalUtils
+{
+    private IntervalUtils()
+    {
+        throw new AssertionError("Suppress default constructor for noninstantiability");
+    }
+
+    public static void throwOnIllegalArguments(long from, long to, long stepBy)
```

Comment:
This should be package private

Reply:
Does `throwOnIllegalArguments` make sense for the method? Or should we name it `checkArguments`

Reply:
It can't be package private because then IntInterval can't use it.

Reply:
I can change the method names.

Reply:
Oh yes. It can’t be package private. Any other way you can think of to not expose the utility as a public method? I think this is an internal implementation and no one other than the library should use it. By making it public we are exposing it. Thoughts?

Reply:
I thought about it.  Couldn't come up with a better alternative yet.  One possibility I thought of was just adding the method to Interval and making it public and then calling from IntInterval to Interval.